### ISSUE TITLE
Deleting a tag prevents updating text

### ISSUE THREAD

**Issue by IceFire23** [standardnotes/forum#1036]
**Describe the bug**
If you delete a tag in standard notes, standard notes will enter a state where I cannot set focus to any text field in the app (as far as I can tell). I can drag-select text and then perform formatting actions on the text, but I can't do anything that requires setting the cursor within the text field.

**To Reproduce**

0. Ensure you have at least one note with text that you will be able to select
1. Add a tag (to be deleted in step 2)
2. Delete tag mentioned in step 1
3. Click on the note referenced in step 0
4. Click in any of the text fields of the selected note (e.g. the title, the tag field, or the body). You can also click on any of the tags in the left pane as well and see the same behavior repro.
5. Notice that the cursor does not appear, and that typing characters or hitting backspace does not update the text at all

**Expected behavior**
I expected that the field would gain focus and my cursor would appear around the area in which I clicked. After the cursor appears I expect to be able to type characters or delete text with backspace.

**Workarounds**
1. Original workaround I found was to close/reopen the app
2. Just found that apparently changing focus to a different app and then re-focusing standard notes is sufficient to resolve issue

**Screenshots**
I had just clicked after the "u" in quick before taking this screenshot. Notice the lack of a cursor.
![NoCursorAfterDeletingTag](https://user-images.githubusercontent.com/35338/87379423-0ecdb900-c545-11ea-9f3d-d2f6855146ac.jpg)

**Desktop (please complete the following information):**
 - OS: Windows 10 v1909 (has happened on older version of OS as well)
 - Standard Notes v 3.4.1 (has happened on older versions of SN as well)

**Additional context**
 - I am using the Bold Editor
 - I am using the Folders extension

**Comment by mobitar** [comment]
I wonder if this happens due to a native system alert appearing? There used to be an issue where after a system alert appeared, text fields would be unfocusable. But come to think about it, no system alert should appear when deleting a tag? In any case, can you confirm if this same issue happens after any kind of alert?

**Comment by IceFire23** [comment]
Yeah, there's no system alert for deleting a tag, at least on Windows. I'm not sure exactly what type of alert you had in mind, but I tried out the "Alarms & Clock" app that I think comes with Windows. I had focus inside of the note, and when a timer went off and a notification appeared, my focus stayed within the app (cursor + ability to type using keyboard). I could even click on the alert to dismiss it and focus stayed within the app.

**Comment by mobitar** [comment]
Sorry, I meant an alert within the app itself. Some ideas:

- Click the Updates top level menu, then Check for update. It should show an alert telling you you're on the latest version.
- Try trashing a note. It will ask you to confirm. Then dismiss that, and see if same issue occurs.

**Comment by IceFire23** [comment]
Ah, gotcha. Results:

1. **Check for Updates**: Saw the alert telling me it's on the latest version. My cursor did disappear while the alert was active, but I could either close the alert and click back on the text, or directly click on the text while the alert was open. Both worked.
2. **Cancelling trash a note dialog**: Went through all the steps before for trashing a note, except cancelled the dialog instead of accepting. Ran into the same problems I had when I trashed the note.

So maybe it is related to the native system alerts you mentioned? Or maybe only the modal dialogs, since the "Check for Updates" dialog was modeless and that worked fine.

**Comment by mobitar** [comment]
In that case it might have something to do with our custom alert. I'll transfer this to the desktop app and assign it to our desktop dev to investigate when he can.
